cmd/build/v2/build.go

```diff
@@ -167,56 +180,92 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 
 			buildSvcInfo := buildManifest[svcToBuild]
 
+			// create the meta pointer and append it to the analytics slice
+			meta := analytics.NewImageBuildMetadata()
+			buildsAnalytics = append(buildsAnalytics, meta)
+
+			meta.Name = svcToBuild
+			meta.RepoURL = bc.Config.GetAnonymizedRepo()
+
+			repoHashDurationStart := time.Now()
+			repoCommit := bc.Config.GetGitCommit()
+			buildHash := getBuildHashFromCommit(buildSvcInfo, repoCommit)
+
+			meta.RepoHash = buildHash
+			meta.RepoHashDuration = time.Since(repoHashDurationStart)
+
 			// We only check that the image is built in the global registry if the noCache option is not set
 			if !options.NoCache && bc.Config.IsCleanProject() {
+
 				imageChecker := getImageChecker(buildSvcInfo, bc.Config, bc.Registry)
-				if imageWithDigest, isBuilt := imageChecker.checkIfCommitHashIsBuilt(options.Manifest.Name, svcToBuild, buildSvcInfo); isBuilt {
-					oktetoLog.Information("Skipping build of '%s' image because it's already built for commit %s", svcToBuild, bc.Config.GetGitCommit())
+				cacheHitDurationStart := time.Now()
```

Comment:
I think `cacheHitDurationStart` should be time to check if some hash is on the available registries but here there is more logic before inside `checkIfBuildHashIsBuilt` func. I would start tracking it in the loop that checks if the digest is in the possible registries

Reply:
i see your point but the process of checking if we already have the image at any registry is running the loop of the different options, the whole step of the build is to get the references and iterate to see if they are at each registry.